### PR TITLE
feat: Add 1.24 support

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

**Reason for Change**:

Some tests in kubernetes/kubernetes which rely on aks-engine to provision clusters have started to fail due to the removal of some kubelet flags. This PR adds support for aks-engine to build 1.24 clusters and removes the deprecated kubelet flags.

Note: There has not been any builds of kubernetes v1.24 yet. The versions referenced in this PR are just a placeholder so we can build clusters from kubernetes/kubernetes at HEAD

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/defaults-kubelet.go [hunk 1]
```diff
@@ -315,6 +315,13 @@ func removeKubeletFlags(k map[string]string, v string) {
 			delete(k, key)
 		}
 	}
+
+	// Remove dockershim related flags in v1.24 and up
+	if common.IsKubernetesVersionGe(v, "1.24.0-alpha") {
+		for _, key := range []string{"--image-pull-progress-deadline", "--network-plugin"} {
+			delete(k, key)
```
Comment (jsturtevant) [inline pkg/api/defaults-kubelet.go:322]: was going to ask if these should these removed or added back in if less than 1.24, but from above it looks like the pattern is to remove the flags

**Review by jsturtevant — APPROVED** [review]

File: pkg/api/defaults-kubelet.go (hunk 1 above)
Comment (marosset) [reply pkg/api/defaults-kubelet.go:322]: Yea, I had the same question and was following previous examples.

**Review by jackfrancis — APPROVED** [review]
lgtm

**Pull request merged**